Disable Sphinx builds on GHA under Python 3.6

Since [about 2021-05-09](https://github.com/iiasa/ixmp/actions/runs/824595459), the "Test documentation build using Sphinx" step of the "pytest" CI workflow is failing—with messages like the following, and *only* on Python 3.6 jobs (not Python 3.8).

This is just a preliminary check that the docs *can* be built. ReadTheDocs is used to actually build & serve them, and that runs on Python 3.7. So it's not important to check whether the docs can be built on Python 3.6 in particular. Therefore the fix should just be to disable this step for this Python version.

```
make: Entering directory '/home/runner/work/ixmp/ixmp/doc'
Running Sphinx v4.0.0
making output directory... done
checking for /home/runner/work/ixmp/ixmp/doc/references.bib in bibtex cache... not found
parsing bibtex file /home/runner/work/ixmp/ixmp/doc/references.bib... parsed 4 entries
[autosummary] generating autosummary for: api-backend.rst, api-model.rst, api-r.rst, api.rst, bibliography.rst, contributing.rst, contributor_license.rst, file-io.rst, index.rst, install.rst, notice.rst, reporting.rst, tutorials.rst, whatsnew.rst
loading intersphinx inventory from https://docs.dask.org/en/stable/objects.inv...
[…]
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 14 source files that are out of date
updating environment: [new config] 14 added, 0 changed, 0 removed
reading sources... [  7%] api

Exception occurred:
  File "/opt/hostedtoolcache/Python/3.6.13/x64/lib/python3.6/inspect.py", line 2381, in _signature_from_callable
    raise ValueError('callable {!r} is not supported by signature'.format(obj))
ValueError: callable operator.itemgetter(3) is not supported by signature
The full traceback has been saved in /tmp/sphinx-err-mkufc3yc.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
make: Leaving directory '/home/runner/work/ixmp/ixmp/doc'
make: *** [Makefile:20: html] Error 2
Error: Process completed with exit code 2.
```

--- a/.github/workflows/pytest.yaml
+++ b/.github/workflows/pytest.yaml
@@ -8,7 +8,6 @@ on:
   schedule:
   - cron: "0 5 * * *"
 
-
 env:
   # See description in lint.yml
   depth: 100
@@ -23,19 +22,14 @@ jobs:
         - windows-latest
         python-version:
         - "3.6"  # Earliest version supported by ixmp
-        - "3.8"  # Latest version testable on GitHub Actions
-
-        # Commented: binary wheels also not available (as of 2020-10-22) for
-        # recently-released Python 3.9.
-        # TODO monitor https://pypi.org/project/[PKG]/#files for the packages
-        #      mentioned below; enable once wheels for "cp39" are available.
-        # - '3.9'  # Latest version / latest supported by ixmp
+        - "3.8"
+        - "3.9"  # Latest release / latest supported by ixmp
 
         # For development versions of Python, compiled binary wheels are not
         # available for some dependencies, e.g. llvmlite, numba, numpy, and/or
         # pandas. Compiling these on the job runner requires a more elaborate
         # build environment, currently out of scope for the ixmp project.
-        # - '3.10.0-alpha.1'  # Development version
+        # - "3.10.0-alpha.1"  # Development version
 
         exclude:
         # See https://github.com/iiasa/ixmp/issues/360

--- a/.github/workflows/pytest.yaml
+++ b/.github/workflows/pytest.yaml
@@ -40,7 +40,7 @@ jobs:
         exclude:
         # See https://github.com/iiasa/ixmp/issues/360
         - os: windows-latest
-          python-version: '3.6'
+          python-version: "3.6"
 
       fail-fast: false
 
@@ -141,6 +141,7 @@ jobs:
       shell: Rscript {0}
 
     - name: Test documentation build using Sphinx
+      if: matrix.python-version != '3.6'
       run: make --directory=doc html
 
     - name: Upload test coverage to Codecov.io

